### ISSUE TITLE
FlatList not properly typed

### ISSUE THREAD

**Issue by MoSattler** [GeekyAnts/NativeBase#4469]
`FlatList`'s `renderItem` does not infer the type from the `data` prop. On the other hand, the regular react-native's flatlets does.

```typescript
<FlatList
  data={["abc"]}
  renderItem={({ item })= > item} // React-native's inferred `item` to be `string`. In native-base it's `any`
/>
```

Same goes for other props like `keyExtractor`.

**Comment by Viraj-10** [comment]
Hi @MoSattler,
Thanks for reporting the issue.
It is fixed now and PR is created. #4478 
It will be included in upcoming release. 😄

**Comment by Viraj-10** [comment]
This issue is fixed in Version 3.3.0 of NativeBase
